mitodl/open-discussions
# Update course-search-utils to fix routing issue

#### Pre-Flight checklist

  - [x] Code is tested (in course-search-utils)

#### What are the relevant tickets?
#3727 

#### What's this PR do?
This PR updates the version of course-search-utils and passes our router's history object to course search utils.

#### How should this be manually tested?

First, **verify the bug on rc:**
1. In a new browser tab, paste `https://discussions-rc.odl.mit.edu/infinite/fields/philosophy/` into the URL bar. **Ensure this is the page that loads the infinite bundle.**
2. Click "Infinite Corridor" in the upper left to go back to homepage 
3. Click the search arrow to go to search page
4. Click a facet on the left hand side.
5. **Observe the bug:** The URL in top reads `https://discussions-rc.odl.mit.edu/infinite/fields/philosophy/?type=course`... The URL pathname is wherever you loaded the bundle, not the search page.

Now, **test install fix:**
1. Check out this branch
2. In a local copy of [course-search-utils](https://github.com/mitodl/course-search-utils)
    ```sh
    git checkout cc/history
    yarn pack
    # move the tgz file to your open discussions repo. If the repos as siblings, then:
    cp mitodl-course-search-utils-v1.8.0.tgz ../open-discussions/mitodl-course-search-utils-v1.8.0.tgz
    ```
3. In the open-discussions repo:
    ```sh
    docker compose run --rm watch bash
    yarn install
    ```
4. Now repeat the "Observe the bug" steps on your local instance with updated course-search-utils. The URL should update correctly now.
5. Also check that `/learn/search` works the same: starting from `/learn` (not /learn/search!) navigate within the app to search and click a facet.

#### Any background context you want to provide?
- https://github.com/mitodl/course-search-utils/pull/36
